lib/karafka/active_job/consumer.rb

```diff
@@ -5,20 +5,91 @@ module ActiveJob
     # This is the consumer for ActiveJob that eats the messages enqueued with it one after another.
     # It marks the offset after each message, so we make sure, none of the jobs is executed twice
     class Consumer < ::Karafka::BaseConsumer
+      # Defaults for consuming
+      # The can be updated by using `#karafka_options` on the job
+      DEFAULTS = {
+        timeout: 300000,          # (5 minutes)
+        max_poll_interval: 300000 # (5 minutes) default max.poll.interval.ms
+      }.freeze
+
       # Executes the ActiveJob logic
       # @note ActiveJob does not support batches, so we just run one message after another
       def consume
         messages.each do |message|
-          ::ActiveJob::Base.execute(
-            # We technically speaking could set this as deserializer and reference it from the
-            # message instead of using the `#raw_payload`. This is not done on purpose to simplify
-            # the ActiveJob setup here
-            ::ActiveSupport::JSON.decode(message.raw_payload)
-          )
-
-          mark_as_consumed(message)
+          # Decode as JSON so we can get the Job Class (Required for finding karafka_options)
+          decoded_message_payload = ::ActiveSupport::JSON.decode(message.raw_payload)
+
+          begin
+            # Pause the consumer for this message partition - so we can poll without receiving
+            # new records
+            client.pause_without_reversal(topic.name, message.partition)
+
+            # Any thread exceptions will be re-raised in the main thread.
+            Thread.abort_on_exception = true
```

Comment:
this is set by default in karafka processing threads

Reply:
Would you prefer this is removed from here? May be worth keeping in case the implementation is changed in karafka processing threads?